Serialization issue with Keycloak 26.4.1

The events plugin has begun displaying WARN with KC version 26.4.1 when sending an event.
Unsure if also present in 26.4.0.

```
Oct 20 16:46:57 keycloak kc.sh[9379]: 2025-10-20 16:46:57,790 WARN  [io.phasetwo.keycloak.events.RunnableTransaction] (executor-thread-
2) Error running Runnable: java.lang.IllegalStateException: Event can't be cloned because of serialization issue.
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.phasetwo.keycloak.events.WebhookSenderEventListenerProvider.clone(WebhookSenderEven
tListenerProvider.java:333)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.phasetwo.keycloak.events.WebhookSenderEventListenerProvider.lambda$processEvent$4(W
ebhookSenderEventListenerProvider.java:155)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.jav
a:1939)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151
)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.ja
va:174)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.phasetwo.keycloak.events.WebhookSenderEventListenerProvider.lambda$processEvent$5(W
ebhookSenderEventListenerProvider.java:153)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.models.utils.KeycloakModelUtils.lambda$runJobInTransaction$1(KeycloakMode
lUtils.java:347)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakMod
elUtils.java:461)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.ja
va:346)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:336)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.phasetwo.keycloak.events.WebhookSenderEventListenerProvider.processEvent(WebhookSenderEventListenerProvider.java:141)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.phasetwo.keycloak.events.WebhookSenderEventListenerProvider.lambda$onEvent$1(WebhookSenderEventListenerProvider.java:85)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.phasetwo.keycloak.events.RunnableTransaction.lambda$commitImpl$0(RunnableTransaction.java:27)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.lang.Iterable.forEach(Iterable.java:75)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.phasetwo.keycloak.events.RunnableTransaction.commitImpl(RunnableTransaction.java:24)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.models.AbstractKeycloakTransaction.commit(AbstractKeycloakTransaction.java:46)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.services.DefaultKeycloakTransactionManager.lambda$commitWithTracing$1(DefaultKeycloakTransactionManager.java:170)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.tracing.NoopTracingProvider.trace(NoopTracingProvider.java:59)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.tracing.NoopTracingProvider.trace(NoopTracingProvider.java:69)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.services.DefaultKeycloakTransactionManager.commitWithTracing(DefaultKeycloakTransactionManager.java:169)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.services.DefaultKeycloakTransactionManager.lambda$commit$0(DefaultKeycloakTransactionManager.java:146)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.tracing.NoopTracingProvider.trace(NoopTracingProvider.java:59)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.tracing.NoopTracingProvider.trace(NoopTracingProvider.java:69)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.services.DefaultKeycloakTransactionManager.commit(DefaultKeycloakTransactionManager.java:121)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.services.resources.admin.UserResource.updateUser(UserResource.java:241)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.services.resources.admin.UserResource$quarkusrestinvoker$updateUser_c4d9b11bd2b4f2348fa10c8d4ebc91a1ff824340.invoke(Unknown Source)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:183)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:645)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at java.base/java.lang.Thread.run(Thread.java:1583)
Oct 20 16:46:57 keycloak kc.sh[9379]: Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "resourceId" (class io.phasetwo.keycloak.representation.ExtendedAdminEvent), not marked as ignorable (13 known properties: "operationTyp
e", "details", "error", "realmId", "time", "uid", "resourcePath", "realmName", "id", "resourceType", "representation", "authDetails", "
type"])
Oct 20 16:46:57 keycloak kc.sh[9379]:  at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column:
 1530] (through reference chain: io.phasetwo.keycloak.representation.ExtendedAdminEvent["resourceId"])
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(Deserializ
ationContext.java:1180)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDese
rializer.java:2244)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1823)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDes
erializerBase.java:1801)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeseriali
zer.java:308)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:169)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(Defau
ltDeserializationContext.java:342)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4971)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3887)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3855)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at org.keycloak.util.JsonSerialization.readValue(JsonSerialization.java:82)
Oct 20 16:46:57 keycloak kc.sh[9379]:         at io.phasetwo.keycloak.events.WebhookSenderEventListenerProvider.clone(WebhookSenderEven
tListenerProvider.java:328)
Oct 20 16:46:57 keycloak kc.sh[9379]:         ... 46 more
```

Using Keycloak-events v0.49

--- a/pom.xml
+++ b/pom.xml
@@ -34,9 +34,9 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <main.java.package>io.phasetwo.keycloak.events</main.java.package>
     <junit.version>5.11.2</junit.version>
-    <keycloak.version>26.3.1</keycloak.version>
-    <keycloak-admin-client.version>26.0.6</keycloak-admin-client.version>
-    <lombok.version>1.18.38</lombok.version>
+    <keycloak.version>26.4.1</keycloak.version>
+    <keycloak-admin-client.version>26.0.7</keycloak-admin-client.version>
+    <lombok.version>1.18.42</lombok.version>
     <auto-service.version>1.1.1</auto-service.version>
     <ossrh.url>https://ossrh-staging-api.central.sonatype.com</ossrh.url>
   </properties>
@@ -134,7 +134,7 @@
       <plugin> <!-- pretty up the code using google java standards `mvn fmt:format` -->
         <groupId>com.spotify.fmt</groupId>
         <artifactId>fmt-maven-plugin</artifactId>
-        <version>2.25</version>
+        <version>2.29</version>
       </plugin>
     </plugins>
   </build>

--- a/src/main/java/io/phasetwo/keycloak/representation/ExtendedAdminEvent.java
+++ b/src/main/java/io/phasetwo/keycloak/representation/ExtendedAdminEvent.java
@@ -152,4 +152,10 @@ public class ExtendedAdminEvent extends AdminEvent {
   public ResourceType getResourceType() {
     return super.getResourceType();
   }
+
+  @Override
+  @JsonProperty(value = "resourceId", access = JsonProperty.Access.READ_ONLY)
+  public String getResourceId() {
+    return super.getResourceId();
+  }
 }

